.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,28 @@
+name: Unit Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'kubernetes/*.go'
+  workflow_dispatch:
+
+jobs:
+  unit_test:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+    - name: Set up Go
+      uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
+      with:
+        go-version-file: 'go.mod'
+    - name: Run unit tests
+      env: 
+       TESTARGS: -count 1 -run '^Test[^Acc]'
```

Comment:
This would prevent a test called `TestALegitimateThing` being run 😝 

Do a negative look aheads (i.e. `^Test(?!Acc)`) work with `go test`?

Reply:
It turns out that Go doesn't support PCRE format. 🙁 I disabled verbose output and removed the `-run` option from Makefile, in this case, only `FAIL` tests will be printed out, I think that should be ok for unit tests.

Reply:
Effectively this regex is selecting every test that doesn't start with `TestA` or `Testc`. I think if we can't use lookaheads it'd be good to simplify the Regex to `^Test[^A]` so it's clearer what it's doing (and the resulting pitfalls)